apps/champions/test/battle_test.exs

```diff
@@ -745,6 +745,152 @@ defmodule Champions.Test.BattleTest do
+              apply_effects_to:
+                TestUtils.build_apply_effects_to_mechanic(%{
+                  effects: [
+                    TestUtils.build_effect(%{
+                      executions: [
+                        %{
+                          type: "Heal",
+                          attack_ratio: 1,
+                          energy_recharge: 0
+                        }
+                      ]
+                    })
+                  ],
+                  targeting_strategy: %{
+                    type: "self"
+                  }
+                })
+            }
+          ],
+          cooldown: skills_cooldown * @miliseconds_per_step - 1
+        })
+
+      {:ok, heal_character} =
+        TestUtils.build_character(%{
+          name: "Self Heal Character",
+          basic_skill: heal_basic_skill_params,
+          ultimate_skill: TestUtils.build_skill(%{name: "Self Heal Empty Skill"}),
+          base_attack: 5,
+          base_health: 10
+        })
+        |> Characters.insert_character()
+
+      {:ok, heal_unit} = TestUtils.build_unit(%{character_id: heal_character.id}) |> Units.insert_unit()
+      {:ok, heal_unit} = Units.get_unit(heal_unit.id)
+
+      deal_damage_basic_skill_params =
+        TestUtils.build_skill(%{
+          name: "DealDamage Skill for Self Test",
+          mechanics: [
+            %{
+              trigger_delay: 0,
+              apply_effects_to:
+                TestUtils.build_apply_effects_to_mechanic(%{
+                  effects: [
+                    TestUtils.build_effect(%{
+                      executions: [
+                        %{
+                          type: "DealDamage",
+                          attack_ratio: 1,
+                          energy_recharge: 50,
+                          delay: 0
+                        }
+                      ]
+                    })
+                  ]
+                })
+            }
+          ],
+          cooldown: skills_cooldown * @miliseconds_per_step - 1
+        })
+
+      # Create enemy unit
+      {:ok, heal_damage_character} =
+        TestUtils.build_character(%{
+          name: "DealDamage Character for Self Test",
+          basic_skill: deal_damage_basic_skill_params,
+          ultimate_skill: TestUtils.build_skill(%{name: "Self Heal Empty Skill 2"}),
+          base_attack: 5
+        })
+        |> Characters.insert_character()
+
+      {:ok, damage_unit_for_heal} =
+        TestUtils.build_unit(%{character_id: heal_damage_character.id}) |> Units.insert_unit()
+
+      {:ok, damage_unit_for_heal} = Units.get_unit(damage_unit_for_heal.id)
+
+      # Battle is timeout, since every time ally unit takes damage, it heals itself
+      assert "timeout" ==
+               Champions.Battle.Simulator.run_battle([heal_unit], [damage_unit_for_heal], maximum_steps: maximum_steps).result
+    end
+
+    test "Self Damage", %{target_dummy: target_dummy} do
+      maximum_steps = 10
```

Comment:
```suggestion
      self_damage_cooldown = 2
      maximum_steps = (self_damage_cooldown + 1) * 4
```

Reply:
This suggestion is wrong as it doesn't replace the other "self_damage_cooldown", so it will result in a warning :C you can implement it by hand